site/content/en/docs/Guides/feature-stages.md

```diff
@@ -0,0 +1,145 @@
+  this happens, we will provide instructions for migrating to the next version. This may require deleting, editing,
+  and re-creating API objects. The editing process may require some thought. This may require downtime for
+  applications that rely on the feature.
+* Recommended for only non-business-critical uses because of potential for incompatible changes in subsequent releases.
+  If you have multiple clusters that can be upgraded independently, you may be able to relax this restriction.
+
+> Note: Please do try `Beta` features and give feedback on them! After they exit beta, it may not be practical for us
+> to make more changes.
+
+### Stable
+
+A `Stable` feature means:
+
+* The feature is enabled and the corresponding feature gate no longer exists.
+* Stable versions of features will appear in released software for many subsequent versions.
+
+## Feature Stage Indicators
+
+There are a variety of features with Agones, how can we determine what stage each feature is in?
+
+Below are indicators for each type of functionality that can be used to determine the feature stage for a given aspect
+of Agones.
+
+### Custom Resource Definitions (CRDs)
+
+This refers to Kubernetes resource for Agones, such as `GameServer`, `Fleet` and `GameServerAllocation`.
+
+#### New CRDs
+
+For new resources, the stage of the resource will be indicated by the `apiVersion` of the resource.
+
+For example: `apiVersion: "agones.dev/v1"` is a `stable` resource, `apiVersion: "agones.dev/v1beta1"` is a `beta`
+ stage resource, and `apiVersion: "agones.dev/v1alpha1"` is an `alpha` stage resource.
+
+#### New CRD attributes
+
+`alpha` and `beta` attributes will have a corresponding `alpha` or `beta` parent element in their configuration to
+delineate their feature stage.
+
+If no such parent exists, this attribute is a `stable` feature.
+
+For example, if we were to add a hypothetical `alpha` configuration option of `timeoutSeconds` to the `GameServer`
+ resource, it would be configured like so:
+ 
+```yaml
+apiVersion: "agones.dev/v1"
+kind: GameServer
+metadata:
+  generateName: "simple-udp-"
+spec:
+  alpha: # this is the alpha feature block
+    timeoutSeconds: 30
+  ports:
+  - name: default
+    portPolicy: Dynamic
+    containerPort: 7654
+  template:
+    spec:
+      containers:
+      - name: simple-udp
+        image: gcr.io/agones-images/udp-server:0.15
+``` 
+
+As resource attributes progress through their stages, there may be breaking changes, as backward conversion between
+ `alpha`, `beta` and `stable` positioning in the resource configuration may not be guaranteed.
+
+### Agones Game Server SDK
+
+Any `alpha` or `beta` Game Server SDK functionality will be a subpackage of the `sdk` package. For example
+, functionality found in a `sdk.alphav1` package should be considered at the `alpha` feature stage.
+
+Only experimental functionality will be found in any `alpha` and `beta` SDK packages, and as such may change as 
```

Comment:
This implies that you would import both the alpha and stable SDKs to start using the alpha sdk functions. I'm not actually sure how that would work in practice (do you have to call `Connect` twice?). Does the alpha package piggyback on the stable one somehow?

Reply:
Yes, you would need to explicitly import the alpha package to use it. But I feel like from an implementation detail we would hide the gRPC details from the user.

From a gRPC technical details, I posted some details on how to do this technically on #1043

> Aha, so I found the right stackoverflow post:
https://stackoverflow.com/questions/52634217/access-multiple-grpc-services-over-the-same-connection
> 
> This shows we can both register multiple services with the same endpoint, and also configure a singular connection with multiple clients - at least with Go.
>
> Even if we end up having 2 connections to the same local endpoint, I don't think it's a drastic issue?

If we had `alpha/`beta` be a superset of `stable`, would we demark a section of sdk.proto for alpha/beta functionality, and add/remove from there as needed? (and basically hide this from the user via our published SDK? This would have some complications for anyone generating a SDK, I would assume?) 

Not to say I'm 100% against a superset, it was just my original viewpoint. I'm not completely married to the idea - maybe a superset is easier for us/the end user.